### PR TITLE
fix(ruby/rubygems)

### PR DESCRIPTION
**Pull request by jhheider**
Ruby is pulling in system gmp

### CODE REVIEW

**Comment by jhheider** [comment]
@mxcl any ideas about this? https://github.com/teaxyz/pantry.core/actions/runs/4189616375/jobs/7262753189

Basically, it appears that ruby-lang on darwin isn't finding openssl; The library is present, the gem is in `lib/ruby/gems/3.2.0/gems` (which isn't our default_dir, but rather the install), and the test for ruby succeeds in using `gem` to install something, but the test in rubygems.org ... doesn't. Oh, and it works locally, of course.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -58,7 +58,7 @@ jobs:
           prefix: /opt
 
       - name: sanitize macOS runners
-        if: matrix.platform.os == 'macos-11'
+        if: fromJson(needs.get-platform.outputs.build-os) == 'macos-11'
```
Comment (mxcl) [inline .github/workflows/build.yml:61]: yuck

**Comment by mxcl** [comment]
I'm currently working on this and other things related to rubygems.

I don’t know what causes it but it seems you fixed it here? I worry that your solution is hard coding a path into something though.

**Comment by davdroman** [comment]
@jhheider @mxcl I believe that's the same problem as seen on https://github.com/teaxyz/cli/issues/374 too.

File: .github/workflows/build.yml (hunk 1 above)
Comment (jhheider) [reply .github/workflows/build.yml:61]: I theoretically agree, but the self-hosted runner identifiers are `string[]` rather than `string`.

Comment (mxcl) [reply .github/workflows/build.yml:61]: yeah not complaining at you, more at GitHub for forcing this

Comment (jhheider) [reply .github/workflows/build.yml:61]: coders like to code, right? who cares about complexity? ;)

**Comment by jhheider** [comment]
rebasing to absorb mxcl's fixes; this now only fixes cleaning up gha runners (which we're not currently using), and prevents linking in system gmp.

**Comment by mxcl** [comment]
I'm still confused about this one. It may crop up again. Let's keep this open for now.

**Pull request merged**